synapse/federation/transport/server/federation.py

```diff
@@ -549,8 +549,7 @@ async def on_POST(
 
 
 class FederationGetMissingEventsServlet(BaseFederationServerServlet):
-    # TODO(paul): Why does this path alone end with "/?" optional?
-    PATH = "/get_missing_events/(?P<room_id>[^/]*)/?"
+    PATH = "/get_missing_events/(?P<room_id>[^/]*)"
```

Comment:
Should that `*` be a `+`? Presumably room ids must be nonempty?

Reply:
Looking at a few other examples, for identifiers we just do `*` and rely on validating them after the fact, so using `+` would be 'inconsistent'. Not sure it makes much difference either way; maybe we should also validate that they start with `!` but I don't know.

Reply:
I think if it was a `+` then the servlet matching code wouldn't even match to this request and you would get a generic `M_UNRECOGNIZED_REQUEST` error. Hopefully by being more graceful in matching we can give a better error about a missing parameters or something?

Reply:
Sounds like we should leave this alone!